files/zh-cn/learn/javascript/building_blocks/functions/index.md

```diff
@@ -127,132 +120,195 @@ myFunction();
+浏览器的内置字符串 [replace()](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String/replace) 函数需要两个参数：在主字符串中查找的子字符串，以及用以下替换该字符串的子字符串：
 
 ```js
-function() {
-  alert('hello');
-}
+const myText = "I am a string";
+const newString = myText.replace("string", "sausage");
 ```
 
-这个函数叫做**匿名函数**——它没有函数名！它也不会自己做任何事情。你通常将匿名函数与事件处理程序一起使用，例如，如果单击相关按钮，以下操作将在函数内运行代码：
+> **备注：** 当你需要指定多个参数时，它们以逗号分隔。
+
+### 可选参数
+
+还应该注意，有时参数不是必需的——你不必指定它们。如果你没有指定某些参数，该函数一般会采用某种默认行为。作为示例，对数组进行操作的 [join()](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array/join) 函数的参数是可选的：
 
 ```js
-var myButton = document.querySelector("button");
+const myArray = ["I", "love", "chocolate", "frogs"];
+const madeAString = myArray.join(" ");
+console.log(madeAString);
+// 返回 'I love chocolate frogs'
 
-myButton.onclick = function () {
-  alert("hello");
-};
+const madeAnotherString = myArray.join();
+console.log(madeAnotherString);
+// 返回 'I,love,chocolate,frogs'
 ```
 
-上述示例将要求 {{htmlelement("button")}} 在页面上提供可用于选择并单击的元素。你在整个课程中已经看到过这种结构了几次，你将在下一篇文章中了解更多信息并在其中使用。
+如果没有包含参数来指定连接/分隔符，默认情况下会使用逗号。
+
+### 默认参数
 
-你还可以将匿名函数分配为变量的值，例如：
+如果你正在编写一个函数，并希望支持可选参数，你可以在参数名称后添加 `=`，然后再添加默认值来指定默认值：
 
 ```js
-var myGreeting = function () {
-  alert("hello");
-};
+function hello(name = "Chris") {
+  console.log(`Hello ${name}!`);
+}
+
+hello("Ari"); // Hello Ari!
+hello(); // Hello Chris!
 ```
 
-现在可以使用以下方式调用此函数：
+## 匿名函数和箭头函数
+
+到目前为止，我们创建了如下函数：
 
 ```js
-myGreeting();
+function myFunction() {
+  alert("hello");
+}
 ```
 
-有效地给变量一个名字;还可以将该函数分配为多个变量的值，例如：
+但是你也可以创建一个没有名称的函数：
 
 ```js
-var anotherGreeting = function () {
+(function () {
   alert("hello");
-};
+});
 ```
 
-现在可以使用以下任一方法调用此函数
+这就是所谓的**匿名函数**，因为它没有名字。当一个函数希望接收另一个函数作为参数时，你经常会看到匿名函数。在这种情况下，函数参数通常作为匿名函数传递。
+
+> **备注：** 这种创建函数的形式也称为函数表达式。与函数声明不同，函数表达式不会被变量提升（hoist）。
+
+### 匿名函数举例
+
+例如，你想在用户输入文本框时运行一些代码。为此，你可以调用文本框的 {{domxref("EventTarget/addEventListener", "addEventListener()")}} 函数。该函数希望你（至少）传给它两个参数：
+
+- 要监听的事件名称，本例中为 {{domxref("Element/keydown_event", "keydown")}}
+- 事件发生时要运行的函数。
+
+当用户按下某个按键时，浏览器将调用你提供的函数，并传递给它一个包含该事件信息的参数，其中包括用户按下的特定按键：
 
 ```js
-myGreeting();
-anotherGreeting();
+function logKey(event) {
+  console.log(`You pressed "${event.key}".`);
+}
+
+textBox.addEventListener("keydown", logKey);
 ```
 
-但这只会令人费解，所以不要这样做！创建方法时，最好坚持下列形式：
+你可以将一个匿名函数传入 `addEventListener()`，而不是定义一个单独的 `logKey()` 函数：
 
 ```js
-function myGreeting() {
-  alert("hello");
-}
+textBox.addEventListener("keydown", function (event) {
+  console.log(`You pressed "${event.key}".`);
+});
 ```
 
-你将主要使用匿名函数来运行负载的代码以响应事件触发（如点击按钮）——使用事件处理程序。再次，这看起来像这样：
+### 箭头函数
+
+如果你传递这样一个匿名函数，你可以使用另一种形式，即箭头函数。你可以写 `(event) =>` 来代替 `function(event)`：
 
 ```js
-myButton.onclick = function () {
-  alert("hello");
-  // 我可以在这里放入
-  // 尽可能多的代码
-};
+textBox.addEventListener("keydown", (event) => {
+  console.log(`You pressed "${event.key}".`);
+});
 ```
 
-> **备注：** 匿名函数也称为函数表达式。函数表达式与函数声明有一些区别。函数声明会进行声明提升（declaration hoisting），而函数表达式不会。
+如果函数只接受一个参数，可以省略参数周围的括号：
 
-## 函数参数
+```js
```

Comment:
Fixes https://github.com/mdn/translated-content/pull/18373/files#r1509250899. Using `-nolint` to prevent linting

````suggestion
```js-nolint
````